src/main/java/edu/hm/hafner/analysis/parser/SimulinkCheckParser.java

```diff
@@ -0,0 +1,135 @@
+package edu.hm.hafner.analysis.parser;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+
+import org.apache.commons.io.input.ReaderInputStream;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.safety.Cleaner;
+import org.jsoup.safety.Safelist;
+import org.jsoup.select.Elements;
+
+import edu.hm.hafner.analysis.IssueBuilder;
+import edu.hm.hafner.analysis.ParsingException;
+import edu.hm.hafner.analysis.ReaderFactory;
+import edu.hm.hafner.analysis.Report;
+import edu.hm.hafner.analysis.Severity;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+/**
+ * A parser for Simulink Check tool by Mathworks results.
+ * Used for .html report files.
+ *
+ * @author Eva Habeeb
+ */
+public class SimulinkCheckParser extends IssuePropertiesParser {
+
+    private static final long serialVersionUID = -8099258658775128275L;
+    static final String WARNING = "div.WarningCheck";
+    static final String FAILED = "div.FailedCheck";
+    static final String NOTRUN = "div.NotRunCheck";
+    static final String INCOMPLETE = "div.IncompleteCheck";
+
+    @SuppressWarnings({"PMD.AvoidCatchingNPE", "PMD.CloseResource"})
+    @SuppressFBWarnings(value = "DCN_NULLPOINTER_EXCEPTION", justification = "throws parsing exception")
+    @Override
+    public Report parse(final ReaderFactory readerFactory) throws ParsingException {
+        try (IssueBuilder issueBuilder = new IssueBuilder()) {
+            Reader reader = readerFactory.create();
+            InputStream targetStream = new ReaderInputStream(reader, readerFactory.getCharset());
+
+            //Create a safe-list with allowed tags and attributes to prevent XSS attacks
+            Safelist safelist = new Safelist();
+            safelist = safelist.basic().addTags("div");
+            safelist = safelist.addAttributes("div", "class", "id");
+            safelist = safelist.addAttributes("span", "class", "id");
```

Comment:
Sorry, but this is not needed anymore *after* Jsoup parsed the document. We simply can delete that part.